src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
-export interface Figure {
-
+function errorParams(...args: number[]): boolean {
+  return args.some((el: number): boolean => {
+    return el <= 0;
+  });
 }
 
-export class Triangle {
+type Form = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Form;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Form = 'triangle';
+
+  static goodTriangle(a: number, b: number, c: number): boolean {
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      return true;
+    }
+
+    return false;
+  }
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (errorParams(a, b, c)) {
+      throw new Error("Error: sides can't be less than 0");
+    }
+
+    if (Triangle.goodTriangle(a, b, c)) {
+      throw new Error('Error: this triangle is impossible');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(semiPerimeter
+      * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
Optional: You do almost the same calculation 3 times. Better create helper function for this purpose.